HobknobClientNet/FeatureToggleCache.cs

```diff
@@ -53,32 +49,19 @@ private bool UpdateCache(out Exception exception)
             }
             catch (Exception ex)
             {
-                if (CacheUpdateFailed != null)
-                {
-                    CacheUpdateFailed(this, new CacheUpdateFailedArgs(ex));
-                }
-
-                exception = ex;
+                CacheUpdateFailed?.Invoke(this, new CacheUpdateFailedArgs(ex));
```

Comment:
Can we keep with C#5 syntax for the moment. I don't fancy going to update MSBuild on all our CI environments at the moment :)